src/link/MachO.zig

```diff
@@ -3656,19 +3656,121 @@ pub fn updateDecl(self: *MachO, module: *Module, decl: *Module.Decl) !void {
     try self.updateDeclExports(module, decl, decl_exports);
 }
 
+fn getMatchingSectionDecl(self: *MachO, decl: *Module.Decl) !MatchingSection {
+    const code = decl.link.macho.code.items;
+    const alignment = decl.ty.abiAlignment(self.base.options.target);
+    const align_log_2 = math.log2(alignment);
+    const ty = decl.ty;
+    const val = decl.val;
+    const match: MatchingSection = blk: {
+        if (ty.zigTypeTag() == .Fn) {
+            break :blk MatchingSection{
+                .seg = self.text_segment_cmd_index.?,
+                .sect = self.text_section_index.?,
+            };
+        }
+        if (mem.allEqual(u8, code, 0)) {
+            break :blk MatchingSection{
+                .seg = self.data_segment_cmd_index.?,
+                .sect = self.bss_section_index.?,
+            };
+        }
+        if (mem.allEqual(u8, code, 0xaa)) {
```

Comment:
This would make the following code miscompile:

```zig
var x: u16 = 0xaaaa;
```

Reply:
Good point!

Reply:
I cleaned the logic up a bit, but I'm certain is not 100% correct nor finished up yet - I'm sure we will be able to tweak and rework it in tree.